src/github.com/turt2live/matrix-media-repo/media_cache/thumbnail_cache.go

```diff
@@ -133,6 +133,12 @@ func (c *mediaCache) GetRawThumbnail(server string, mediaId string, width int, h
 
 	// At this point we'll try and generate the thumbnail
 	thumb, err = thumbnailSvc.GenerateThumbnail(media, width, height, method, animated)
+	if err != nil && err == errs.ErrMediaNotAnimated {
```

Comment:
This shouldn't be neccasary - the thumbnail code *should* already be considering the animated flag only for animated media, and therefore changing it to `animated=false` if the media is not compatible.

Reply:
yeah, and that is exactly what this does. I throw an error inside the GenerationThumbnail to then set animated to true to re-fetch it. The check that was in place there didn't work, as it didn't re-check cache and thus was trying to re-insert a new, identical record into postgresql resulting in M_UNKOWN

Reply:
I'm not sure this is the right place to fix it then. The cache is supposed to be relatively dumb and forward requests to the service. ThumbnailService should instead keep track of it's changes to the animated flag and call GetThumbnailDirect if the animated flag changes to try and return an already-generated thumbnail instead.

Reply:
Okay, I'll look more into it